src/visitors/transpileCssProp.js

```diff
@@ -18,13 +18,17 @@ export default t => (path, state) => {
   if (!useCssProp(state)) return
   if (path.node.name.name !== 'css') return
 
+  const program = path.findParent(p => t.isProgram(p))
+  const importName =
+    state.customImportName || program.scope.generateUidIdentifier('styled')
+  if (!state.customImportName) state.customImportName = importName
   // Insert require('styled-components') if it doesn't exist yet
-  const { bindings } = path.findParent(p => p.type === 'Program').scope
+  const { bindings } = program.scope
   if (!state.required) {
-    if (!bindings.styled) {
-      state.items.push(
+    if (!bindings[importName]) {
+      program.node.body.push(
```

Comment:
They are correctly transpiled to `_interopRequireDefault(require('styled-components'))` in both the unit tests and my local test app, but I'm not sure if that's just by chance...

Reply:
@mxstbr depends on the setup I think. plugins run before presets, so I guess it'll be okay if someone uses `babel-preset-env`. but if they use the individual plugins for some reason (for example they have their own preset, it might break).

though not sure why you prefer `_interopRequireDefault(require('styled-components'))` since adding `require('styled-components')` is inserting less code and doesn't need any transpilation.

Reply:
I'd prefer to stick to the thing that people are doing before we rewrite their code as possible, if that makes sense.

If folks want to keep `import`s around and not transpile to `require`s, we shouldn't force `require`s on them, right?

Reply:
> If folks want to keep `imports` around and not transpile to `require`s, we shouldn't force requires on them, right?

IMO people use `import` because better tree-shaking. unless there's such an advantage here, I doubt anyone cares how their code looks after transpilation.

So by using `import` we basically provide no extra advantage, require specific conditions to work properly (the commonjs plugin must run after this plugin), and insert extra interop code to the bundle which is unnecessary.

Reply:
>  I doubt anyone cares how their code looks after transpilation.

What if you don't want to transpile down to CommonJS, but you want to keep ESM around (for whatever reason)? We'd basically be forcing the choice on everybody.

Reply:
It's also possible that the user doesn't use ES imports in their app at all only uses commonjs, we'll never be certain. The thing about commonjs is every tool understands it. ES imports require transpilation to be used outside few bundlers. We could also say the same about `const` vs `var` (the plugin inserts `var`, but the user might be using `const`)

Anyway, it's up to you. Just keep in mind that even if we don't need to `requeue` it with the common setup, depending on the babel configuration a `requeue` might be needed to transpile it. So if you want to insert ES import, make sure to `requeue` to be safe.

Reply:
Yeah that makes sense, will have a think about it!

Reply:
Another option is provide 2 macros, one which outputs commonJS require + ES5 compat code, one that outputs imports + ES6 compat code.

```js
// However you want to name it
import 'styled-components/macro'; // ES5 require version
import 'styled-components/esmacro'; // ES6 import version
```
This leaves the choice to the end developer.

Reply:
⚠️The macro itself always outputs es6 imports.

There's 2 use cases.

1. The macro is imported via the main import

```js
import styled from 'styled-components/macro'
// or
const styled = require('styled-components/macro')
```

In this case, the macro will reinject es6 import for styled-components and will set `state.customImportName`, so `bindings[importName]` will always be true. So the code here to add `require('styled-components')` won't be executed.

The output is (both with import and require) :
```js
import _styled from 'styled-components'
```

2. The macro is imported via named imports

```js
import {css} from 'styled-components'
```

In this case, the macro will reinject es6 named imports, won't set `state.customImportName` and `bindings[importName]` will be false, so the code will run and will add `require('styled-components')`.

The output is :
```js
import { css as _css } from 'styled-components'
const _styled = require('styled-components')
```

Reply:
Huh good catch @lucleray. Let's move to `require` across the board? Would that require a patch in the macro? Want to submit a PR for that?

Reply:
It's dangerous to go alone! [take this](https://babeljs.io/docs/en/next/babel-helper-module-imports.html).

Reply:
Thanks @Kovensky ! This is awesome 🙌

Reply:
@Kovensky yessss this is exactly what we need, thank you so much! I'm going to switch this PR to use it and we should also use it in the macro—want to submit a PR for that @lucleray?

Reply:
@mxstbr I'll write a PR for the macro 🕵️

Reply:
AFAIK `sourceType` is only for parse format. It still needs correct plugin ordering for the transform. So we should add a `requeue` after the insert.